data_pipeline_api/file_formats/parameter_file.py

```diff
@@ -0,0 +1,109 @@
+import toml
+from io import TextIOBase
+from enum import Enum
+from numbers import Real
+from typing import Union, Sequence, Dict, Any
+import numpy as np
+from scipy import stats
+
+# ======================================================================================
+# Common
+# ======================================================================================
+
+ParameterComponent = Dict[str, Any]
+Estimate = Real
+Distribution = Union[stats.rv_discrete, stats.rv_continuous]
+Samples = np.ndarray
+
+
+class Type(Enum):
+    point_estimate = "point-estimate"
+    distribution = "distribution"
+    samples = "samples"
+
+
+def read_parameter(file: TextIOBase, component: str) -> ParameterComponent:
+    file.seek(0)
+    return toml.load(file)[component]
+
+
+def write_parameter(file: TextIOBase, component: str, parameter: ParameterComponent):
+    parameter_data = toml.load(file)
+    parameter_data[component] = parameter
+    file.seek(0)
+    file.truncate()
+    toml.dump(parameter_data, file)
+
+
+def read_type(file: TextIOBase, component: str) -> Type:
+    parameter = read_parameter(file, component)
+    return Type(parameter["type"])
+
+
+# ======================================================================================
+# Estimate
+# ======================================================================================
+
+
+def read_estimate(file: TextIOBase, component: str) -> Estimate:
+    parameter = read_parameter(file, component)
+    if Type(parameter["type"]) is Type.point_estimate:
+        # TODO : validate
+        return parameter["value"]
+    else:
+        raise ValueError(f"{parameter['type']} != 'point-estimate'")
+
+
+def write_estimate(file: TextIOBase, component: str, estimate: Estimate):
+    write_parameter(file, component, {"type": "point-estimate", "value": estimate})
+
+
+# ======================================================================================
+# Distribution
+# ======================================================================================
+
+distribution_parsers = {
+    "gamma": lambda data: stats.gamma(a=data["shape"], scale=data["scale"]),
```

Comment:
Not sure if relevant, but the distributions we are actually using are:

- Poisson
- Multinomial and binomial
- Hypergoemetric

Although I don see us reading the parameters of those distributions from this file, as they are sort of dynamically created (I'm considering the number of trials and array of probabilities to be the parameters here). @FiodorG may have better ideas.

Reply:
I'll leave this for now, but I think that we should add required distributions to the spec, and then add them to the implementation.

Reply:
How are we going to read parameters from them though? The array of probabilities and number of trials are calculate dynamically, aren they? How could that be in a file?

Reply:
I don't think I understand the question properly - surely if they are being dynamically created we won't be loading them?

Reply:
> How are we going to read parameters from them though?
The answer to this is that the distribution objects contain the parameters, but I still don't get the second part of what you said.

Reply:
What will we read from the file given that all parameters we are using right now are computed dynamically? This API assumes that you have all the parameters you need in a file before you run your simulation.

Reply:
My understanding is that the intent of the "distribution" type is to represent a complete distribution. If you only have individual parameters then you could either read those separately, as point-estimates, or you could read a complete distribution and ignore some of its parameters.

Reply:
Cool. So I think, at least for now, we are not going to be making use of these in simple network sim.